Src/Support/Google.Apis/Requests/BatchRequest.cs

```diff
@@ -228,6 +229,52 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task EnsureSuccess(HttpResponseMessage result)
+        {
+            if (!result.IsSuccessStatusCode)
+            {
+                RequestError error = null;
+                try
+                {
+                    // Try to parse the error from the current response.
+                    error = await service.DeserializeError(result).ConfigureAwait(false);
+                }
+                catch
+                {
+                    // If for any reason we couldn't parse the error,
+                    // do nothing, we'll end up calling EnsureSuccessStatusCode.
+                }
+
+                try
+                {
+                    // Now that we may have more error detail, let's call EnsureSuccessStatusCode.
+                    // We don't want to introduce breaking changes for users that relied on
+                    // HttpRequestException before, and importantly on its message format which is the only
+                    // way they could access the HttpStatusCode.
+                    result.EnsureSuccessStatusCode();
+                }
+                catch (HttpRequestException original) when (error != null)
+                {
+                    // We only get here if we have meaningful information to add
+                    // to the original exception.
+                    try
+                    {
+                        // We throw here instead of simply creating a new GoogleApiException
+                        // so as to get the StackTrace.
+                        throw new GoogleApiException(service.Name, error.ToString())
+                        {
+                            Error = error,
+                            HttpStatusCode = result.StatusCode
+                        };
+                    }
+                    catch(GoogleApiException extra)
+                    {
+                        throw new HttpRequestException(original.Message, extra);
```

Comment:
Pass in `original.StatusCode` as well?

Reply:
HttpRequestException.StatusCode is [only available from .NET 5.0](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestexception.statuscode?view=net-5.0&viewFallbackFrom=netcore-3.1#System_Net_Http_HttpRequestException_StatusCode).

A lot of people were asking/complaining (SO, etc.) about the exception not having an explicit StatusCode property. The guidance from MS was to call HttpResponseMessage.StatusCode, but that of course is not possible in cases such as ours in which libraries are just hiding the response object and throwing the exception. Probably why they've added it now.